### ISSUE TITLE
unknown block type rotation_policy

### ISSUE THREAD

**Issue by Nielshensen** [hashicorp/terraform-provider-azurerm#22584]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.3

### AzureRM Provider Version

3.65.0

### Affected Resource(s)/Data Source(s)

azurerm_key_vault_key

### Terraform Configuration Files

```hcl
resource "azurerm_key_vault_key" "generated" {
    name         = "generated-certificate"
    key_vault_id = azurerm_key_vault.example.id
    key_type     = "RSA"
    key_size     = 2048

    key_opts = [
        "decrypt",
        "encrypt",
        "sign",
        "unwrapKey",
        "verify",
        "wrapKey",
    ]

    rotation_policy {

    }
}
```

### Debug Output/Panic Output

```shell
error message: unknown block type rotation_policy
```

### Expected Behaviour

When trying to use the azurerm_key_vault_key resource I want to add the rotation_policy. This way I can make sure the key rotates every 90 Days. 

However I'm getting the unknown block type error.

I have looked for documentation but that states that this should work.
I have the latest terraform version as wel as the latest azure provider version.

**Comment by Nielshensen** [comment]
I Have tried to add those attributes but it gives me the same result. 
This is the complete file

`locals {
    current_user_id = coalesce(var.msi_id, data.azurerm_client_config.current.object_id)
}

resource "azurerm_resource_group" "rg" {
    name     = var.resource_group_name
    location = var.region
}

data "azurerm_client_config" "current" {}

resource "random_string" "azurerm_key_vault_name" {
    length = 13
    lower = var.random_string_settings.lower
    numeric = var.random_string_settings.numeric
    special = var.random_string_settings.special
    upper = var.random_string_settings.upper
}

resource "azurerm_key_vault" "vault" {
    location            = azurerm_resource_group.rg.location
    name                = coalesce(var.vault_name, "vault-${random_string.azurerm_key_vault_name.result}")
    resource_group_name = azurerm_resource_group.rg.name
    sku_name            = var.sku_name
    tenant_id           = data.azurerm_client_config.current.tenant_id
    soft_delete_retention_days = 7

    access_policy = {
        tenant_id = data.azurerm_client_config.current.tenant_id
        object_id = local.current_user_id
        key_permissions = [
            "Create",
            "Delete",
            "Get",
            "Purge",
            "Recover",
            "Update",
            "GetRotationPolicy",
            "SetRotationPolicy"
        ]

        secret_permissions = [
            "Set",
        ]
    }
}

resource "random_string" "azurerm_key_vault_key_name" {
    length = 13
    lower = var.random_string_settings.lower
    numeric = var.random_string_settings.numeric
    special = var.random_string_settings.special
    upper = var.random_string_settings.upper
}

resource "azurerm_key_vault_key" "generated" {
    name         = "generated-certificate"
    key_vault_id = azurerm_key_vault.vault.id
    key_type     = "RSA"
    key_size     = 2048

    key_opts = [
        "decrypt",
        "encrypt",
        "sign",
        "unwrapKey",
        "verify",
        "wrapKey",
    ]

    rotation_policy {
        automatic {
            time_before_expiry = "P30D"
        }

        expire_after         = "P90D"
        notify_before_expiry = "P29D"
    }
}
`

And the added attributes shouldn't make a difference since they are optional as you can see in this documentation:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_key.html

**Comment by sinbai** [comment]
@Nielshensen thanks for opening this issue. Per the provided complete file, it seems that the `access_policy = {` should be replaced with `access_policy  {`. The following tf config works fine on my side, could you try with the following config?
```

terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
	  version = "3.65.0"
    }
  }
}


provider "azurerm" {
  features {}
}


resource "azurerm_resource_group" "rg" {
    name     = "22584"
    location = "eastus"
}

data "azurerm_client_config" "current" {}


resource "azurerm_key_vault" "vault" {
    location            = azurerm_resource_group.rg.location
    name                = "acctest20230720"
    resource_group_name = azurerm_resource_group.rg.name
    sku_name            = "standard"
    tenant_id           = data.azurerm_client_config.current.tenant_id
    soft_delete_retention_days = 7

    access_policy {
        tenant_id = data.azurerm_client_config.current.tenant_id
        object_id = data.azurerm_client_config.current.object_id
        key_permissions = [
            "Create",
            "Delete",
            "Get",
            "Purge",
            "Recover",
            "Update",
            "GetRotationPolicy",
            "SetRotationPolicy"
        ]

        secret_permissions = [
            "Set",
        ]
    }
}

resource "azurerm_key_vault_key" "generated" {
    name         = "generated-certificate"
    key_vault_id = azurerm_key_vault.vault.id
    key_type     = "RSA"
    key_size     = 2048

    key_opts = [
        "decrypt",
        "encrypt",
        "sign",
        "unwrapKey",
        "verify",
        "wrapKey",
    ]

    rotation_policy {
        automatic {
            time_before_expiry = "P30D"
        }

        expire_after         = "P90D"
        notify_before_expiry = "P29D"
    }
}
```

**Comment by SeenYang** [comment]
Could you double check on the version that actually running the tf code? We encoutered similar issue and turns out the pipeline runs on 3.33.0 which the feature was released on 3.46.0.

**Comment by rcskosir** [comment]
Thank you for taking the time to raise this! I am going to close this with @sinbai‘s response as an answer. If you have future questions, I suggest using the [Community Resources](https://www.terraform.io/community), such as the [Azure Provider forum](https://discuss.hashicorp.com/c/terraform-providers/tf-azure/34).
